site/content/en/concepts/packaging/_index.md

```diff
@@ -25,4 +25,54 @@ For compatibility with other arbitrary formats, kpt supports generating
 resource configuration data from templates, configuration DSLs, and programs
 using [source functions].
 
+## Composite Packages
+
+A `composite package` is a `kpt` package which contains 1 or more `kpt` packages
+in its directory tree.
+
+Here is an example directory structure of a `composite package`
+
+```sh
+foo # composite package
+├── Kptfile
+├── bar # composite package
+│   ├── Kptfile
+│   ├── baz
+│   │   ├── Kptfile
+│   │   └── cm.yaml
+│   └── deploy.yaml
+└── service.yaml
+```
+
+#### Use cases
+
+1. Package publishers need a way to pick and choose multiple component packages
+   which work well together, create a single `composite package` using them to
+   ship an out of the box application, maintain the package and abstract the
+   details from package consumers. Alternatively, package consumers may [update]
+   individual packages directly from the upstream sources.
+2. Package publishers need a way to enforce parameter values (ex: [setters]) to
```

Comment:
I'm not sure if what we are providing today fits this description. We provide a way to create setters across nested packages, but we don't really do any enforcement.